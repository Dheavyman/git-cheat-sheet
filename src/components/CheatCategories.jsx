import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Masonry from 'react-masonry-component';

import { fetchCheats } from '../actions/cheats';
import CheatCategory from './CheatCategory';

class CheatCategories extends Component{
  render() {
    const { categorizedCheats } = this.props;

    return (
      <div>
        <Masonry>
          {[...categorizedCheats].map(([category, cheats]) => (
            <CheatCategory key={category} category={category} cheats={cheats} />
          ))}
        </Masonry>
      </div>
    );
  }
};

const mapStateToProps = ({ cheatsReducer: { cheats, searchResult, keywords } }) => {
  const cheatsToDisplay = keywords ? searchResult : cheats;
  const categorizedCheats = [...cheatsToDisplay].reduce((accumulator, cheat) => {
    const category = cheat['category'];
    const categoryCheats = accumulator.get(category) || []

    categoryCheats.push(cheat)
    return accumulator.set(category, categoryCheats)
  }, new Map());
  return ({
    categorizedCheats
  });
};

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({ fetchCheats }, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(CheatCategories);
