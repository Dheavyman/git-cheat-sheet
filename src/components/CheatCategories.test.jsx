import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { CheatCategories } from './CheatCategories';

describe('CheatCategories component', () => {
  const key1 = 'cheat category';
  const key2 = 'another category';
  const value1 = [{
    _id: 'cheat id',
    description: 'cheat description',
    command: 'cheat command',
  }];
  const value2 = [{
    _id: 'another cheat id',
    description: 'another cheat description',
    command: 'another cheat command',
  }]
  const categorizedCheats = new Map()
  categorizedCheats.set(key1, value1);
  categorizedCheats.set(key2, value2);
  const props = {
    categorizedCheats
  };
  it('should render component successfully', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<CheatCategories {...props} />);
    const result = renderer.getRenderOutput()

    expect(result).toMatchSnapshot();
  });
});
