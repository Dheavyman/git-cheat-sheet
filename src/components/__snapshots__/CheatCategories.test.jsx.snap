// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheatCategories component should render component successfully 1`] = `
<div>
  <MasonryComponent
    className=""
    disableImagesLoaded={false}
    elementType="div"
    enableResizableChildren={false}
    imagesLoadedOptions={Object {}}
    onLayoutComplete={[Function]}
    onRemoveComplete={[Function]}
    options={Object {}}
    updateOnEachImageLoad={false}
  >
    <CheatCategory
      category="cheat category"
      cheats={
        Array [
          Object {
            "_id": "cheat id",
            "command": "cheat command",
            "description": "cheat description",
          },
        ]
      }
    />
    <CheatCategory
      category="another category"
      cheats={
        Array [
          Object {
            "_id": "another cheat id",
            "command": "another cheat command",
            "description": "another cheat description",
          },
        ]
      }
    />
  </MasonryComponent>
</div>
`;
